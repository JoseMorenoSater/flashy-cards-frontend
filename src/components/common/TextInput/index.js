/**
 *
 * TextInput
 *
 */

import React from "react";
import PropTypes from "prop-types";

const TextInput = ({
  divClassName,
  label,
  inputClassName,
  inputId,
  onBlur,
  onChange,
  defaultValue,
  value,
  inputType
}) => {
  return (
    <div className={divClassName}>
      <label htmlFor="lessonName">{label}</label>
      <input
        id={inputId}
        onChange={onChange}
        value={value}
        defaultValue={defaultValue}
        type={inputType ? inputType : "text"}
        className={inputClassName}
        onBlur={onBlur}
      />
    </div>
  );
};

TextInput.propTypes = {
  divClassName: PropTypes.string,
  label: PropTypes.string,
  inputClassName: PropTypes.string,
  inputId: PropTypes.string,
  onBlur: PropTypes.func,
  defaultValue: PropTypes.string,
  onChange: PropTypes.func,
  value: PropTypes.string,
  inputType: PropTypes.string
};

export default TextInput;
