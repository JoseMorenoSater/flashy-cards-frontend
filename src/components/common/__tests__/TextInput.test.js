import React from "react";
import { create } from "react-test-renderer";
import TextInput from "../TextInput";

describe("TextInput Component", () => {
  it("Should render correctly", () => {
    const TextInputComponent = create(<TextInput />).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
  });
});
