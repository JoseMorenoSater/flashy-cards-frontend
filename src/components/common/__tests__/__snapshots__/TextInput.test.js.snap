// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component Should render correctly 1`] = `
<div
  className={undefined}
>
  <label
    htmlFor="lessonName"
  />
  <input
    className={undefined}
    defaultValue={undefined}
    id={undefined}
    onBlur={undefined}
    onChange={undefined}
    type="text"
    value={undefined}
  />
</div>
`;
